File: src/java/Utils/DiscreteMarginalDistribution.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */

package Utils;

import java.util.HashMap;

/**
 *
 * @author XPS 1645
 */
public class DiscreteMarginalDistribution extends MarginalDistribution
{
    private HashMap<String, Object> myCollection;
    private String dimension;

    public String GetDimensionName()
    {
        return dimension;
    }

    public void SetDimensionName(String dim)
    {
        dimension = dim;
    }

    public DiscreteMarginalDistribution()
    {
        myCollection = new HashMap<String, Object>();
    }

    public int GetCategoryCount()
    {
        return myCollection.size();
    }

    public boolean AddValue(String category, double val)
    {
        if (!myCollection.containsKey(category))
        {
            myCollection.put(category, val);
            return true;
        }
        return true;
    }

    @Override
    public double GetValue(String category)
    {
        if(myCollection.containsKey(category))
        {    
            return Double.parseDouble(myCollection.get(category).toString());
        }
        return 0.00;
    }
}

File: src/c#/Utils/DiscreteMarginalDistribution.cs
/*
 * created by: b farooq, poly montreal
 * on: 22 october, 2013
 * last edited by: b farooq, poly montreal
 * on: 22 october, 2013
 * summary: 
 * comments:
 */

using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace PopulationSynthesis.Utils
{
    class DiscreteMarginalDistribution : MarginalDistribution
    {
        private Hashtable myCollection;
        private string dimension;
        public string GetDimensionName()
        {
            return dimension;
        }
        public void SetDimensionName(string dim)
        {
            dimension = dim;
        }

        public DiscreteMarginalDistribution()
        {
            myCollection = new Hashtable();
        }
        public int GetCategoryCount()
        {
            return myCollection.Count;
        }

        public bool AddValue(string category, double val)
        {
            if (!myCollection.Contains(category))
            {
                myCollection.Add(category, val);
                return true;
            }
            return true;
        }

        public override double GetValue(string category)
        {
            if(myCollection.Contains(category))
            {
                return (double)myCollection[category];
            }
            return 0.00;
        }

    }
}
